psydac/feec/multipatch/plotting_utilities.py

```diff
@@ -0,0 +1,300 @@
+from psydac.fem.basic          import FemField
+from psydac.feec.pull_push     import push_2d_h1, push_2d_hcurl, push_2d_hdiv, push_2d_l2
+
+from mpi4py import MPI
+#==============================================================================
+# some plotting utilities
+# todo (MCP, april 12): merge get_grid_vals_scalar and get_grid_vals_vector into a single function
+def get_grid_vals_scalar(u, etas, mappings_list, space_kind='h1'):  #_obj):
+    # get the physical field values, given the logical field and the logical grid
+    # n_patches = len(domain)
+    n_patches = len(mappings_list)
+    u_vals    = n_patches*[None]
+    for k in range(n_patches):
+        eta_1, eta_2 = np.meshgrid(etas[k][0], etas[k][1], indexing='ij')
+        u_vals[k] = np.empty_like(eta_1)
+        if isinstance(u,FemField):
+            uk_field = u.fields[k]   # todo (MCP): try with u[k].fields?
+        else:
+            # then field is just callable
+            uk_field = u[k]
+
+        if space_kind == 'h1':
+            # todo (MCP): add 2d_hcurl_vector
+            push_field = lambda eta1, eta2: push_2d_h1(uk_field, eta1, eta2)
+        else:
+            push_field = lambda eta1, eta2: push_2d_l2(uk_field, eta1, eta2, mapping=mappings_list[k])
+        for i, x1i in enumerate(eta_1[:, 0]):
+            for j, x2j in enumerate(eta_2[0, :]):
+                u_vals[k][i, j] = push_field(x1i, x2j)
+
+    # u_vals  = np.concatenate(u_vals, axis=1)
+
+    return u_vals
+
+#------------------------------------------------------------------------------
+def get_grid_vals_vector(E, etas, mappings_list, space_kind='hcurl'):
+    # get the physical field values, given the logical field and logical grid
+    n_patches = len(mappings_list)
+    # mappings_list = list(mappings.values())
+    E_x_vals = n_patches*[None]
+    E_y_vals = n_patches*[None]
+    for k in range(n_patches):
+        eta_1, eta_2 = np.meshgrid(etas[k][0], etas[k][1], indexing='ij')
+        E_x_vals[k] = np.empty_like(eta_1)
+        E_y_vals[k] = np.empty_like(eta_1)
+        if isinstance(E,FemField):
+            Ek_field_0 = E[k].fields[0]   # or E.fields[k][0] ?
+            Ek_field_1 = E[k].fields[1]
+        else:
+            # then E field is just callable
+            Ek_field_0 = E[k][0]
+            Ek_field_1 = E[k][1]
+        if space_kind == 'hcurl':
+            # todo (MCP): specify 2d_hcurl_scalar in push functions
+            push_field = lambda eta1, eta2: push_2d_hcurl(Ek_field_0, Ek_field_1, eta1, eta2, mapping=mappings_list[k])
+        else:
+            push_field = lambda eta1, eta2: push_2d_hdiv(Ek_field_0, Ek_field_1, eta1, eta2, mapping=mappings_list[k])
+
+        for i, x1i in enumerate(eta_1[:, 0]):
+            for j, x2j in enumerate(eta_2[0, :]):
+                E_x_vals[k][i, j], E_y_vals[k][i, j] = push_field(x1i, x2j)
+    # E_x_vals = np.concatenate(E_x_vals, axis=1)
+    # E_y_vals = np.concatenate(E_y_vals, axis=1)
```

Comment:
What are these commented lines for? Can we remove them?

Reply:
the same as above, we can remove them

Reply:
done in [4c54406](https://github.com/pyccel/psydac/pull/160/commits/4c5440653d00be8d85aa7c35a2b9c66927667b5a)